Proto file lacking a package can result in processing failure

## Description

This appears to occur when the library attempts to resolve message field types, where the type is non-scalar and not defined within a package.

From what I can tell, this may down to the fact that the [fullyQualifiedName()](https://github.com/lyft/protoc-gen-star/blob/95ca57a4ef989c084ed1a5a58810d2b34a28cf72/name.go#L184) function looks to include the FQN of the parent. The parent can have return FQN of `.` - so this function can return something along the lines of `..Foo`. When [hydrating field types](https://github.com/lyft/protoc-gen-star/blob/95ca57a4ef989c084ed1a5a58810d2b34a28cf72/ast.go#L284), an FQN value derived from the field is passed down - where no package is specified, this will have a value along the lines of `.Foo`. The mismatch results in Failf() being called.

## Steps to reproduce

Create proto file containing no package declaration:

```
$ mkdir testdata/protos/nopackage/
$ cat > testdata/protos/nopackage/foo.proto <<EOF
syntax = "proto3";

message Foo {
    uint64 id = 1;
}

message Bar {
    Foo foo = 1;
}
EOF
```

Make testdata target:

```
$ make testdata
<snip>
# generate using our demo plugin, don't need to go directory at a time
set -e; for subdir in `find ./testdata/protos -type d -mindepth 1 -maxdepth 1`; do \
                protoc -I ./testdata/protos \
                        --plugin=protoc-gen-example=./bin/protoc-gen-example \
                        --example_out="paths=source_relative:./testdata/generated" \
                        `find $subdir -name "*.proto"`; \
        done
expected entity ".Foo" has not been hydrated
--example_out: protoc-gen-example: Plugin failed with status code 1.
make: *** [testdata/generated] Error 1

```

--- a/ast.go
+++ b/ast.go
@@ -2,7 +2,7 @@ package pgs
 
 import (
 	"github.com/golang/protobuf/protoc-gen-go/descriptor"
-	"github.com/golang/protobuf/protoc-gen-go/plugin"
+	plugin_go "github.com/golang/protobuf/protoc-gen-go/plugin"
 )
 
 // AST encapsulates the entirety of the input CodeGeneratorRequest from protoc,

--- a/ast_test.go
+++ b/ast_test.go
@@ -6,7 +6,7 @@ import (
 	"testing"
 
 	"github.com/golang/protobuf/proto"
-	"github.com/golang/protobuf/protoc-gen-go/plugin"
+	plugin_go "github.com/golang/protobuf/protoc-gen-go/plugin"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
@@ -257,4 +257,31 @@ func TestGraph_HydrateFieldType_Group(t *testing.T) {
 
 	assert.Nil(t, g.hydrateFieldType(f))
 	assert.True(t, md.Failed())
+}
+
+func TestGraph_Packageless(t *testing.T) {
+	t.Parallel()
+
+	g := buildGraph(t, "packageless")
+
+	tests := []struct {
+		name        string
+		entityIFace interface{}
+	}{
+		{".RootMessage", (*Message)(nil)},
+		{".RootEnum", (*Enum)(nil)},
+		{".RootMessage.field", (*Field)(nil)},
+		{".RootEnum.VALUE", (*EnumValue)(nil)},
+		{".RootMessage.NestedMsg", (*Message)(nil)},
+		{".RootMessage.NestedEnum", (*Enum)(nil)},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ent, ok := g.Lookup(tc.name)
+			assert.True(t, ok)
+			assert.NotNil(t, ent)
+			assert.Implements(t, tc.entityIFace, ent)
+		})
+	}
 }

--- a/file.go
+++ b/file.go
@@ -45,7 +45,6 @@ type file struct {
 }
 
 func (f *file) Name() Name                                  { return Name(f.desc.GetName()) }
-func (f *file) FullyQualifiedName() string                  { return "." + f.desc.GetPackage() }
 func (f *file) Syntax() Syntax                              { return Syntax(f.desc.GetSyntax()) }
 func (f *file) Package() Package                            { return f.pkg }
 func (f *file) File() File                                  { return f }
@@ -56,6 +55,13 @@ func (f *file) MapEntries() (me []Message)                  { return nil }
 func (f *file) SourceCodeInfo() SourceCodeInfo              { return f.SyntaxSourceCodeInfo() }
 func (f *file) SyntaxSourceCodeInfo() SourceCodeInfo        { return f.syntaxInfo }
 func (f *file) PackageSourceCodeInfo() SourceCodeInfo       { return f.packageInfo }
+
+func (f *file) FullyQualifiedName() string {
+	if pkg := f.desc.GetPackage(); pkg != "" {
+		return "." + pkg
+	}
+	return ""
+}
 
 func (f *file) Enums() []Enum {
 	return f.enums

--- a/testdata/graph/packageless/packageless.proto
+++ b/testdata/graph/packageless/packageless.proto
@@ -1,0 +1,17 @@
+syntax="proto3";
+
+// no package declaration!
+
+message RootMessage {
+    RootEnum field = 1;
+
+    message NestedMsg {}
+
+    enum NestedEnum {
+        VALUE = 0;
+    }
+}
+
+enum RootEnum {
+    VALUE = 0;
+}

